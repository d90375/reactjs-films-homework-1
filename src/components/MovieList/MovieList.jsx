import React from 'react';
import PropTypes from 'prop-types';
import MovieItem from '../MovieItem';

import styles from './MovieList.scss';

const MovieList = ({
  movies, fetchTrailer, removeDetailsInfo, setMoviesCondition,
}) => {
  const movieItems = movies.map((film) => (
    <MovieItem
      key={film.id}
      film={film}
      fetchTrailer={fetchTrailer}
      removeDetailsInfo={removeDetailsInfo}
      setMoviesCondition={setMoviesCondition}
    />
  ));

  return (
    <div className={styles.container}>
      {movieItems.length ? movieItems
        : <p className={styles.noResults}>No results</p>}
    </div>
  );
};

MovieList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
  fetchTrailer: PropTypes.func.isRequired,
  removeDetailsInfo: PropTypes.func.isRequired,
  setMoviesCondition: PropTypes.func.isRequired,
};

export default MovieList;
