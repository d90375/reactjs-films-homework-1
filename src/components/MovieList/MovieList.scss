.container {
  display: grid;
  grid-template-columns: repeat(auto-fill, 250px);
  grid-gap: 20px;
  justify-content: center;
  padding-top: 20px;
  border-top: 1px solid rgb(207, 207, 207);

  @media (min-width: 390px) and (max-width: 659px) {
    grid-template-columns: repeat(auto-fill, 90%);
  }
}

.noResults {
  padding-top: 40px;
  font-size: 22px;
  text-transform: uppercase;
  color: rgb(73, 73, 73);
}
