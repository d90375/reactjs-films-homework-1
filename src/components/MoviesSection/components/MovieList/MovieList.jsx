import React from 'react';
import PropTypes from 'prop-types';
import MovieItem from '../MovieItem';
// import WatchNowWindow from '../WatchNowWindow';
// import ViewInfoWindow from '../ViewInfoWindow';

import styles from './MovieList.scss';

const MovieList = ({ movies }) => {
  const movieItems = movies.map((film) => (
    // <div key={film.id} className={styles.movie}>
    <MovieItem film={film} />

    //   <WatchNowWindow key={`watch${id}`} film={film} />
    //   { film.viewInfo ? <ViewInfoWindow key={`view${id}`} film={film} /> : null }

    // </div>
  ));

  return (
    <div className={styles.container}>
      {movieItems}
    </div>
  );
};

MovieList.propTypes = {
  movies: PropTypes.arrayOf.isRequired,
};

export default MovieList;
