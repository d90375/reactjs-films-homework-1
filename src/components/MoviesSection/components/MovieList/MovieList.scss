.container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0 -10px;
}

.movie {
  position: relative;

  &:hover > div:nth-child(2) {
    display: flex;
  }
}
