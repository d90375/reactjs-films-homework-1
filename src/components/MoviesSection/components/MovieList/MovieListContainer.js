import { connect } from 'react-redux';
import { getTranding } from '../../../../modules/trending/trendingSelector';
import MovieList from './MovieList';
import { getSearch, getSearchQuery } from '../../../../modules/search/searchSelector';

const mapStateToProps = (state) => ({
  tranding: getTranding(state),
  search: getSearch(state),
  searchQuery: getSearchQuery(state),
});

export default connect(
  mapStateToProps,
)(MovieList);