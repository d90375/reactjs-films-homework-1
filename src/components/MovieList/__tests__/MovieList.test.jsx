import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';
import MovieList from '../MovieList';

describe('MovieList tests', () => {
  describe('MovieList render', () => {
    it('MovieList renders correctly', () => {
      const movies = [{ id: 1 }, { id: 2 }];
      const mockCallBack = jest.fn();
      const renderer = new ShallowRenderer();
      renderer.render(<MovieList fetchTrailer={mockCallBack} movies={movies} />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });

    it('MovieList renders correctly', () => {
      const movies = [];
      const mockCallBack = jest.fn();
      const renderer = new ShallowRenderer();
      renderer.render(<MovieList fetchTrailer={mockCallBack} movies={movies} />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  });

  /* describe('onClick function', () => {
    it('onClick callback function called', () => {
      const mockCallBack = jest.fn();

      const tree = create(
        <MovieList fetchByFilter={mockCallBack} filter="Trending" condition="Trending" onClick={mockCallBack}>Trending</MovieList>,
      );

      const button = tree.root.findByProps({ type: 'button' });
      const e = {
        target: {
          dataset: {
            filter: 'Trending',
          },
        },
      };

      button.props.onClick(e);

      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack).toHaveBeenCalledWith('Trending');
    });
  }); */
});
