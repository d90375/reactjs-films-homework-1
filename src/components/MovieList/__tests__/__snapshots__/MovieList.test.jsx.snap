// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList tests MovieList render MovieList renders correctly 1`] = `
<div
  className="container"
>
  <MovieItem
    fetchTrailer={[MockFunction]}
    film={
      Object {
        "id": 1,
      }
    }
  />
  <MovieItem
    fetchTrailer={[MockFunction]}
    film={
      Object {
        "id": 2,
      }
    }
  />
</div>
`;

exports[`MovieList tests MovieList render MovieList renders correctly 2`] = `
<div
  className="container"
>
  <p
    className="noResults"
  >
    No results
  </p>
</div>
`;
